system/libraries/Session/drivers/Session_redis_driver.php

```
@@ -272,7 +313,7 @@ public function close()
 		if (isset($this->_redis))
 		{
 			try {
-				if ($this->_redis->ping() === '+PONG')
+				if ($this->_redis->ping() === $this->_ping_success)
```

Glad to see this change has made it in, gave me a lot of trouble until I realized today what was happening before seeing the pull request.

Should there be a log message if the ping does not return as expected? It becomes a silent failure that's very hard to track down, and causes most sessions to stay locked.

---

That is right. It should be improved.